include/fastdds/rtps/participant/RTPSParticipant.h

```diff
@@ -283,6 +287,46 @@ class RTPS_DllAPI RTPSParticipant
     bool ignore_reader(
             const GUID_t& reader_guid);
 
+    /**
+     * Enables the monitor service in this RTPSParticipant.
+     *
+     * @return RETCODE_OK if the monitor service could be correctly enabled.
+     * @return RETCODE_ERROR if the monitor service could not be enabled properly.
+     *
+     * @note Not supported yet. Currently returns RETCODE_UNSUPPORTED
+     */
+    ReturnCode_t enable_monitor_service() const;
+
+    /**
+     * Disable the monitor service in this RTPSParticipant. Does nothing if the service was not enabled before.
+     *
+     * @return RETCODE_OK if the monitor service could be correctly disabled.
+     * @return RETCODE_NOT_ENABLED if the monitor service was not previously enabled.
+     * @return RETCODE_ERROR if the service could not be properly disabled.
+     *
+     * @note Not supported yet. Currently returns RETCODE_UNSUPPORTED
+     */
+    ReturnCode_t disable_monitor_service() const;
+
+    /**
+     * Enables the internal monitor service in this RTPSParticipant.
+     *
+     * @param wdata Pointer to the WriterProxyData object.
+     * @param rdata Pointer to the ReaderProxyData object.
+     * @param [out] reason On return will specify the reason of failed matching (if any).
+     * @param [out] incompatible_qos On return will specify all the QoS values that were incompatible (if any).
+     *
+     * @return RETCODE_OK if qos are compatible.
+     * @return RETCODE_ERROR if qos are incompatible. Check reason and incompatible_qos.
+     *
+     * @note Not supported yet. Currently returns RETCODE_UNSUPPORTED
+     */
+    ReturnCode_t check_compatible_qos(
```

Comment:
See that this method is just the same as `EDP::valid_matching` (passing pointers instead of references :wink: 
 I think I know now where the doxygen documentation has come from :sweat_smile: ). I think the name should be kept